src/Client.js

```diff
@@ -344,14 +358,28 @@ class Client extends EventEmitter {
      * Request authentication via pairing code instead of QR code
      * @param {string} phoneNumber - Phone number in international, symbol-free format (e.g. 12025550108 for US, 551155501234 for Brazil)
      * @param {boolean} showNotification - Show notification to pair on phone number
+     * @param {number} intervalMs - The interval in milliseconds on how frequent to generate pairing code (WhatsApp default to 3 minutes)
      * @returns {Promise<string>} - Returns a pairing code in format "ABCDEFGH"
      */
-    async requestPairingCode(phoneNumber, showNotification = true) {
-        return await this.pupPage.evaluate(async (phoneNumber, showNotification) => {
-            window.AuthStore.PairingCodeLinkUtils.setPairingType('ALT_DEVICE_LINKING');
-            await window.AuthStore.PairingCodeLinkUtils.initializeAltDeviceLinking();
-            return window.AuthStore.PairingCodeLinkUtils.startAltLinkingFlow(phoneNumber, showNotification);
-        }, phoneNumber, showNotification);
+    async requestPairingCode(phoneNumber, showNotification = true, intervalMs = 180000) {
+        return await this.pupPage.evaluate(async (phoneNumber, showNotification, intervalMs) => {
+            const getCode = async () => {
+                window.AuthStore.PairingCodeLinkUtils.setPairingType('ALT_DEVICE_LINKING');
+                await window.AuthStore.PairingCodeLinkUtils.initializeAltDeviceLinking();
+                return window.AuthStore.PairingCodeLinkUtils.startAltLinkingFlow(phoneNumber, showNotification);
+            };
+            if (window.codeInterval) {
+                clearInterval(window.codeInterval); // remove existing interval
+            }
+            window.codeInterval = setInterval(async () => {
+                if (window.AuthStore.AppState.state != 'UNPAIRED' && window.AuthStore.AppState.state != 'UNPAIRED_IDLE') {
+                    clearInterval(window.codeInterval);
+                    return;
+                }
+                window.onCodeReceivedEvent(await getCode());
+            }, intervalMs);
```

Comment:
@MobCode100 Is there a way we could get rid of this custom interval and use the WhatsApp one? Maybe there is an event that triggered once the code is expired and we can subscribe to this event?

Reply:
Is there anyway to find out? Using inspect element or some tool?

Reply:
@alechkos  According to my research, there is a timeout callback:
```
        e = d("useWAWebTimeout").useTimeout(function() {
            var a;
            d("WALogger").LOG(m());
            (a = d("WAWebAltDeviceLinkingApi").getCurrentMarker()) == null ? void 0 : a.addPoint("code_ttl_expire");
            B()
        }, 3.25 * d("WATimeUtils").MINUTE_MILLISECONDS);
```
However the callback is only created on the `Enter code on phone` page. We will have to navigate to that page to benefit from it but this defeats the purpose of `webpack-exodus` branch.

The simplest way to get rid of this custom interval is to match the default interval `3.25 * d("WATimeUtils").MINUTE_MILLISECONDS`. The 3 minute value seems to be hard-coded as far as I can find.

Reply:
> The simplest way to get rid of this custom interval is to match the default interval `3.25 * d("WATimeUtils").MINUTE_MILLISECONDS`. The 3 minute value seems to be hard-coded as far as I can find.

Should this feature follow the default whatsapp timer and remove the custom interval parameter?